.circleci/config.yml

```diff
@@ -13,9 +13,83 @@ jobs:
 
     steps:
       - checkout
+      # -batch is necessary to make sure sbt doesn't stall waiting for user input in case of certain failures
+      - run: echo '-batch' > .sbtopts
 
-      - run: sbt -batch update
+      - run: sbt update
+      - run: sbt +compile
+      - run: sbt +test:compile
+      - run: sbt +test
+      - persist_to_workspace:
+          root: '~'
+          paths: ['*']
 
-      - run: sbt -batch +test:compile
+  smoke-test:
+    machine:
+      # https://circleci.com/docs/2.0/configuration-reference/#available-machine-images
+      image: ubuntu-1604:201903-01
+    working_directory: ~/infrastructure-as-types
+    steps:
+      - run:
+          name: Install kind
+          command: |
+            curl -fLo ./kind "https://kind.sigs.k8s.io/dl/v0.8.1/kind-Linux-amd64"
+            chmod +x ./kind
+            sudo mv ./kind /usr/local/bin/kind
+      - run:
+          name: Install kubectl
+          command: |
+            curl -fLo ./kubectl https://storage.googleapis.com/kubernetes-release/release/v1.18.0/bin/linux/amd64/kubectl
+            chmod +x ./kubectl
+            sudo mv ./kubectl /usr/local/bin/kubectl
+
+      - run:
+          name: Create cluster
+          command: |
+            kind create cluster
+            mkdir -p ~/.kube/
+            kind get kubeconfig > ~/.kube/config
+
+      # No checkout, just attaching the workspace from the build job.
+      - attach_workspace:
+          at: '~'
+      - run:
+          name: Deploy GuestBook
+          environment:
+            IAT_KUBE_CONTEXT: kind-kind
+          no_output_timeout: 1m
+          command: |
+            sbt "project iat-examples" "runMain com.virtuslab.iat.examples.GuestBook"
+      - run:
+          name: Wait for GuestBook frontend pods to spin up
+          command: |
+            kubectl wait --namespace guestbook --for condition=ready --selector app=guestbook --timeout 5m pods
```

Comment:
@pawelprazak do you think some option in infra-as-types that would act like helm upgrade --wait (wait until deployed resources are ready) would be useful? This approach with kubectl wait is kinda hard to scale (what if it's not just 3 services but a lot more?)

Reply:
sure, makes sense, feel free to add PR

Reply:
#9 added as for now